src/screens/Products/AddProduct.js

```diff
@@ -0,0 +1,310 @@
+      productTimesSold: 0,
+      focusedInputs: {title: false, desc: false, price: false, sku: false, quantity: false, time: false},
+      adding: false
+    }
+
+    this.pickImage = this.pickImage.bind(this)
+    this.createProduct = this.createProduct.bind(this)
+    this.verifyFields = this.verifyFields.bind(this)
+  }
+
+  pickImage() {
+    ImagePicker.openPicker({
+      width: 300,
+      height: 300,
+      cropping: true
+    }).then(image => {
+      this.setState({productMainPicture: image.path})
+    });
+  }
+
+  verifyFields() {
+    if(!this.state.productMainCategory)
+      this.props.displayError("! The required field Main Category is empty.")
+    else if(!this.state.productSubCategory)
+      this.props.displayError("! The required field Subcategory is empty.")
+    else if(!this.state.productMainPicture)
+      this.props.displayError("! The required field Main Picture is empty.")
+    else if(!this.state.productTitle)
+      this.props.displayError("! The required field Title is empty.")
+    else if(!this.state.productDescription)
+      this.props.displayError("! The required field Product Description is empty.")
+    else if(!this.state.productGender)
+      this.props.displayError("! The required field Gender is empty.")
+    else if(!this.state.productStandardPrice)
+      this.props.displayError("! The required field Standard Price is empty.")
+    else if(!this.state.productSellerSKU)
+      this.props.displayError("! The required field ? is empty.")
+    else if(!this.state.productQuantity)
+      this.props.displayError("! The required field Quantity is empty.")
+    else if(!this.state.productTime)
+      this.props.displayError("! The required field Production Time is empty.")
+
+    return this.state.productMainCategory && this.state.productSubCategory
+      && this.state.productMainPicture && this.state.productTitle && this.state.productDescription
+      && this.state.productGender && this.state.productStandardPrice && this.state.productSellerSKU
+      && this.state.productQuantity && this.state.productTime
+  }
+  
+  createProduct() {
+    if(this.verifyFields()) {
+      this.setState({adding: true})
+      this.props.createProduct({
+        mainCategory: this.state.productMainCategory.trim(),
+        subCategory: this.state.productSubCategory.trim(),
+        mainPicture: this.state.productMainPicture.trim(),
+        title: this.state.productTitle.trim(),
+        description: this.state.productDescription.trim(),
+        gender: this.state.productGender.trim(),
+        standardPrice: this.state.productStandardPrice,
+        sellerSKU: this.state.productSellerSKU.trim(),
+        quantity: this.state.productQuantity,
+        productionTime: this.state.productTime,
+        timesSold: this.state.productTimesSold
+      }, this.props.navigation.getParam('currentUID')).then(() => {
+        this.setState({adding: false})
+        this.props.navigation.goBack()
+      })
+    }
+  }
+
+  render() {
```

Comment:
This is a pretty massive list of categories and sub-categories, possibly shorten this or omit subcategories for simplicity. This is used in the drop down menu's so functionality is used but possibly have this in a different file etc.